packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -269,34 +234,48 @@ def _get_provider(self, provider_id: Optional[str]) -> BaseProvider:
 
         return self.providers[provider_id]
 
-    @magic_arguments()
-    @argument('model_id',
-                help="""Model to run, specified as a model ID that may be
-                optionally prefixed with the ID of the model provider, delimited
-                by a colon.""")
-    @argument('-f', '--format',
-                choices=["code", "html", "image", "json", "markdown", "math", "md", "text"],
-                nargs="?",
-                default="markdown",
-                help="""IPython display to use when rendering output. [default="markdown"]""")
-    @argument('-r', '--reset',
-                action="store_true",
-                help="""Clears the conversation transcript used when interacting
-                with an OpenAI chat model provider. Does nothing with other
-                providers.""")
-    @argument('prompt',
-                nargs='*',
-                help="""Prompt for code generation. When used as a line magic, it
-                runs to the end of the line. In cell mode, the entire cell is
-                considered the code generation prompt.""")
+    def handle_help(self, _: HelpArgs):
+        with click.Context(cell_magic_parser, info_name="%%ai") as ctx:
+            click.echo(cell_magic_parser.get_help(ctx))
+        click.echo('-' * 78)
+        with click.Context(line_magic_parser, info_name="%ai") as ctx:
+            click.echo(line_magic_parser.get_help(ctx))
+
+    def handle_list(self, args: ListArgs):
+        return TextOrMarkdown(
+            self._ai_list_command_text(args.provider_id),
+            self._ai_list_command_markdown(args.provider_id)
+        )
+
     @line_cell_magic
     def ai(self, line, cell=None):
-        # parse arguments
-        args = parse_argstring(self.ai, line)
-        if cell is None:
-            prompt = ' '.join(args.prompt)
+        raw_args = line.split(' ')
+        if cell:
+            args = cell_magic_parser(raw_args, prog_name="%%ai", standalone_mode=False)
         else:
-            prompt = cell
+            args = line_magic_parser(raw_args, prog_name="%ai", standalone_mode=False)
+
+        if args == 0:
+            # this happens when `--help` is called on the root command, in which
+            # case we want to exit early.
+            return
+
+        if args.type == "help":
+            return self.handle_help(args)
+        if args.type == "list":
+            return self.handle_list(args)
+        
+        # hint to the IDE that this object must be of type `RootArgs`
+        args: CellArgs = args
+
+        if not cell:
+            raise CellMagicError(
+                """[0.8+]: To invoke a language model, you must use the `%%ai`
```

Comment:
```suggestion
                """To invoke a language model, you must use the `%%ai`
```

Do we need to indicate the module version? Because we're still before version 1.0, I think an occasional logical change is expected.

Reply:
I think it's OK to keep this as-is; 7 characters is a small price to pay for giving users a hint as to why their line magics no longer work.